tooling/nargo_cli/src/cli/check_cmd.rs

```diff
@@ -34,6 +35,8 @@ pub(crate) struct CheckCommand {
 
 pub(crate) fn run(args: CheckCommand, config: NargoConfig) -> Result<(), CliError> {
     let toml_path = get_package_manifest(&config.program_dir)?;
+    let lock = Lock::lock(toml_path.clone());
```

Comment:
Have you considered doing the lock somewhere [here](https://github.com/noir-lang/noir/blob/cb0628b86a0e37db9d62d84e340c1a9ae254c340/tooling/nargo_cli/src/cli/mod.rs#L128-L129), so the lock/unlock pattern doesn't have to be repeated in every command? It would apply the ones that as you say don't produce artefacts, but would also make things simpler elsewhere; maybe they could use [lock_shared](https://docs.rs/fs2/latest/fs2/trait.FileExt.html#tymethod.lock_shared) so they don't block each other.

Reply:
Oh, good point! I'll move it there.

> maybe they could use [lock_shared](https://docs.rs/fs2/latest/fs2/trait.FileExt.html#tymethod.lock_shared) so they don't block each other

I'm not sure about this. The file already exists and nobody writes to it... and if they don't block each other then it kind of defeats the purpose of this PR? But I must be misunderstanding something.

Reply:
Sorry, I meant that whatever writes artifacts would use lock_exclusively, while the other commands that just read artifacts use lock_shared. That way they can’t be mutated while they are used by read-only commands

Reply:
Oooh... I see! Makes sense, I'll make this change.

Reply:
I just realized some commands don't need a lock at all, like `new` and `init`. Then LSP runs in the background... I don't know if it would be bad for that to lock things because then you couldn't run some commands on the command line.

Given that this PR was about solving the issue of running multiple `execute` commands in parallel in CI without issues, maybe only exclusively locking in a few commands and just that is good enough for now.